packages/lib/src/storybook/storybookArgTypes.ts

```diff
@@ -21,61 +21,97 @@ export interface StoryArgType extends InputType {
+  type: BaseArgTypes,
+): (description?: string, config?: Record<string, any>) => StoryArgType;
+function argType(
+  type: ArgTypeWithOptions,
+): (
+  options: Array<any>,
+  description?: string,
+  config?: Record<string, any>,
+) => StoryArgType;
+function argType(
+  type: BaseArgTypes | ArgTypeWithOptions,
+): (...args: any) => StoryArgType {
+  if (['radio', 'check', 'select', 'multi-select'].includes(type)) {
+    return (
+      options: Array<any>,
+      description?: string,
+      config?: Record<string, any>,
+    ) =>
+      ({
+        description,
+        options,
+        control: { type, ...config },
+      } as StoryArgType);
+  } else
+    return (description?: string, config?: Record<string, any>) =>
+      ({
+        description,
+        control: { type, ...config },
+      } as StoryArgType);
+}
+
+export const StorybookArgTypes = {
+  Boolean: argType('boolean'),
+  Number: argType('number'),
+  Range: argType('range'),
+  Select: argType('select'),
+  Radio: argType('radio'),
+  Check: argType('check'),
+  Text: argType('text'),
+  None: { control: 'none' },
+} as const;
+
+/**
+ * Default configurations for storybook arg types
+ */
+export const defaultStorybookArgTypes: Record<string, StoryArgType> = {
+  baseFontSize: StorybookArgTypes.Radio(
+    [14, 16],
+    'The base font size passed to the LeafyGreenProvider that wraps the component',
+  ),
+  updatedBaseFontSize: StorybookArgTypes.Radio(
+    [13, 16],
+    'The base font size passed to the LeafyGreenProvider that wraps the component. Uses the updated font size values for Euclid Circular A.',
+  ),
+  darkMode: StorybookArgTypes.Boolean('Render the component in dark mode.'),
+  children: StorybookArgTypes.Text('Element rendered inside the component'),
+  as: StorybookArgTypes.Select(
+    IntrinsicElements,
+    'The component will be rendered in HTML as the element selected here',
+  ),
 } as const;
 
-/** By default we set argType to `control:none`
+/**
+ * By default we set argType to `control:none`
  * for props we don't want to control in Storybook
  */
 export const storybookExcludedArgTypes: Record<string, StoryArgType> = {
-  'aria-controls': { control: 'none' },
-  'aria-describedby': { control: 'none' },
-  'aria-label': { control: 'none' },
-  'aria-labelledby': { control: 'none' },
-  className: { control: 'none' },
-  id: { control: 'none' },
-  onBlur: { control: 'none' },
-  onCancel: { control: 'none' },
-  onChange: { control: 'none' },
-  onClear: { control: 'none' },
-  onClick: { control: 'none' },
-  onClose: { control: 'none' },
-  onConfirm: { control: 'none' },
-  onDismiss: { control: 'none' },
-  onFilter: { control: 'none' },
-  onSubmit: { control: 'none' },
-  portalClassName: { control: 'none' },
-  portalContainer: { control: 'none' },
-  popoverZIndex: { control: 'none' },
-  ref: { control: 'none' },
-  scrollContainer: { control: 'none' },
-  usePortal: { control: 'none' },
+  'aria-controls': StorybookArgTypes.None,
```

Comment:
wondering if we want to be more selective with what props we choose to demo, rather than having an exclude list